angular2/src/app/inbox/inbox.component.ts

```diff
@@ -1,42 +1,34 @@
 import {Component, OnInit} from '@angular/core';
 import * as Talk from "talkjs";
+import { TalkJsService } from '../talkjs.service';
 
 @Component({
   selector: 'app-inbox',
-  template: `<div id='talkjs-inbox-container'></div>`
+  template: `
+    <div *ngIf="currentUser" id='talkjs-inbox-container'>Loading....</div>
+    <div *ngIf="!currentUser">Please log in first.</div>
+  `
 })
 export class InboxComponent implements OnInit {
-  constructor() {}
+  private session: Promise<Talk.Session>;
+  private currentUser: string;
 
-  ngOnInit() {
-    Talk.ready.then(() => {
-      const me = new Talk.User({
-        id: 'my-id-for-user',
-        name: 'John Doe',
-        configuration: 'buyer'
-      });
-
-      const talkSession = new Talk.Session({
-        appId: 'Hku1c4Pt',
-        me
-      });
+  constructor(private talkJs: TalkJsService) {}
 
-      const conversation = talkSession.getOrCreateConversation("order_66");
-      conversation.setParticipant(me);
-      conversation.setAttributes({
-        subject: "Star Wars"
-      });
-      
-      const inbox = talkSession.createInbox({selected: conversation});
+  ngAfterViewInit() {
+    if(!this.session) {
+      return;
+    }
+    this.session.then(session => {
+      const inbox = session.createInbox();
```

Comment:
I think to make it consistent with home.component `inbox` should be a class member of this component.

Reply:
Agreed